src/DynamicRouter.php

```diff
@@ -223,8 +192,17 @@ public function match($pathinfo)
 
         $request = Request::create($pathinfo);
         if ($this->eventDispatcher) {
+            $eventName = Events::PRE_DYNAMIC_MATCH;
             $event = new RouterMatchEvent();
-            $this->eventDispatcher->dispatch(Events::PRE_DYNAMIC_MATCH, $event);
+
+            // LegacyEventDispatcherProxy exists in Symfony >= 4.3
+            if (class_exists(LegacyEventDispatcherProxy::class)) {
```

Comment:
lets add a private dispatchEvent method to the router class to extract this logic to a separate place.

for a moment i was wondering if we need the old way when we are with symfony 5 routing, but probably we should, as we can't prevent routing 5 from being installed together with event dispatcher 4